daemon.go

```diff
@@ -85,6 +85,18 @@ func (d *Daemon) getNodeWithToken(label string, token *Token) (*Node, error) {
 	return node, nil
 }
 
+func (d *Daemon) usingNodeWithToken(label string, token *Token,
+	f func(*Node) error) error {
+	d.Lock()
+	defer d.Unlock()
+	node, err := d.getNodeWithToken(label, token)
+	if err != nil {
+		return err
+	}
+	f(node)
+	return nil
```

Comment:
Should change this to `return f(node)`; otherwise we're silencing errors from f.

Reply:
good catch, in my head I was thinking that `f(node)` will "raise" an error rather than simply "return" error